src/tree-key-cache.ts

```diff
@@ -156,4 +179,110 @@ export class TreeKeyCache<T> {
 			}
 		}
 	}
+
+	/**
+	 * Set all the given values of the informed tree
+	 * @param tree The tree with the values to be set. It need to start from root node. Any node you don't want to change, just leave it undefined
+	 * @param createValue Callback for leaf creation when it does not exist
+	 */
+	async *fullTreeSet(
+		tree: Tree<T>,
+		createValue: () => T,
+	): AsyncIterable<FullSetItem<T>> {
+		const iterable = treePreOrderBreadthFirstSearch(tree);
+		const { keyLevelNodes } = this.options;
+		const deserialize: (stringified: string) => T =
+			this.options.deserialize ?? defaultDeserialize;
+		const serialize: (stringified: T) => string =
+			this.options.serialize ?? defaultSerialize;
+		let currentSerialized: string | undefined;
+		for (const breadthNode of iterable) {
+			const { level } = breadthNode;
+			if (level > keyLevelNodes) {
+				break;
+			}
+			const { key } = breadthNode;
+			const chainedKey = buildKey(breadthNode);
+			if (level === keyLevelNodes) {
+				yield* this.saveFullTreeValue(
+					breadthNode,
+					chainedKey,
+					serialize,
+					deserialize,
+					createValue,
+				);
+				continue;
+			}
+			const value = breadthNode.value;
+			currentSerialized = (await this.storage.get(chainedKey))?.toString();
+			const old = getStep(
+				deserialize,
+				currentSerialized,
+				key,
+				level,
+				createValue,
+			);
+			const oldValue = old.value;
+			const newValue = value ?? oldValue;
+			const step: FullSetItem<T> = {
+				...old,
+				oldValue,
+				value: newValue,
+			};
+			yield step;
+			const serialized = serialize(newValue);
+			if (currentSerialized !== serialized) {
+				await this.storage.set(chainedKey, serialized);
+			}
+		}
+	}
+
+	private async *saveFullTreeValue(
+		breadthNode: TraversalItem<T>,
+		chainedKey: string,
+		serialize: (payload: T) => string,
+		deserialize: (stringified: string) => T,
+		createValue: () => T,
+	): AsyncIterable<FullSetItem<T>> {
+		const { treeRef } = breadthNode;
+		const serializedTree = await this.storage.get(chainedKey);
+		const rootTree: Tree<string> = serializedTree
+			? JSON.parse(serializedTree)
+			: {};
+		let currentTree: Tree<string> | undefined = rootTree;
+		const stack = [];
+		let changed = false;
+		const rootItem = getFullSetItem(currentTree, breadthNode, deserialize);
+		yield rootItem.node;
+		changed = checkFullSetItemChange(rootItem, changed, serialize, currentTree);
+		for (const depthNode of treePreOrderDepthFirstSearch(treeRef)) {
```

Comment:
The iteration where the pairs that fall into a tree structure must be made with a depth-first search, because every in-depth node starting from the tree structure point will be saved in the same cache key,  so, the iteration in that branch must be completed to, after that, the persistence can be made